src_code/src/controllers/services/Inventory/GetAllModels.js

```diff
@@ -0,0 +1,65 @@
+/** Initialize neccessary modules */
+const responseBuilder = require("../../../utils/interfaces/IResponseBuilder");
+const db = require("../../../configurations/database/DatabaseConfigurations");
+
+/**
+ * Retrieves information about all equipment models and their respective equipment counts.
+ *
+ * @param {Object} res - Express response object for sending HTTP responses.
+ * @returns {Object} An HTTP response containing information about equipment models and their counts.
+ * If successful, returns a JSON response with status 200 and an array of model information.
+ * If no models are found, returns a JSON response with status 400 and an error message.
```

Comment:
Why status code 400? I've noted this in the previous PR: I don't understand why some "not founds" give 400 and others give 404.

Reply:
The 404 and 400 typically indicates error. The range of 400 is the error range server response to client. 404 indicate that models is not found in the system. Some return 400 and some return 404 depends on how you implement the validation. if you take a look at front end, if we consider its 200 as OK for fetching object and data then 400 will be the error message, same for 503 and 500.

Reply:
In other words, 400 to response with a "backend defined message". error 404 is already means not found, and if you want to return a message with error 404 related to it through a backend defined message, it is not making any sense. The error status we put here depend on the situation and meaningful of the error.

Reply:
I'm not sure if I fully understand what you are trying to get here. Are you saying 400 also includes "item doesn't exist"? This seems to tell otherwise:
https://stackoverflow.com/questions/49811464/the-difference-between-the-400-and-404-http-errors
https://developer.mozilla.org/en-US/docs/Web/HTTP/Status#client_error_responses

Reply:
The way we return and implement 400 and 404 is different. In the case, we return 404 for the get endpoint meanwhile in validation we return 400. More than that, the range error 400 is still the same.

Reply:
It depends on how we want it to be. For example, in some situation, you may want to return 200 when there is nothing found in a list for a get request. But we are using the logic if there is nothing found, we return 404 for get. and if failed validation return 400. There are some cases we return 404 in validation, it is too emphasized the entity we want to retrieve is important and not found.

Reply:
You might came across of why we are using 503 but not 500? It depends on how we want to construct our response in order to make it work on the front end side.

Reply:
404 also sometimes return when client making request to a non-exist endpoint.

Reply:
So, you're saying that for get endpoints, we would want to return 404 if nothing is found. From what I can tell, GetAllModels is a get endpoint, so according to the logic I'm seeing here, I don't understand why it should return 400 if nothing is found.

Reply:
I don't know what to tell to make this clear for you. Those response makes connection to communicate with the client side...

Reply:
Okay, so how exactly is it doing that? Why not have 404 implemented so that it can also communicate with the client?

Reply:
I changed it to 404 as you requested. Please try to read the code base and understand the logic. I have tried all my explanation, sorry if my explanation doesnt help. If in the future needed, we can change back 400.😄

Reply:
Sounds good. Thanks for trying to explain to me! This may be something that we will definitely need to define clearly, but for now, I think this should suffice.